File: python/semantic_kernel/prompt_template/const.py
# Copyright (c) Microsoft. All rights reserved.

from typing import Literal

KERNEL_TEMPLATE_FORMAT_NAME: Literal["semantic-kernel"] = "semantic-kernel"
HANDLEBARS_TEMPLATE_FORMAT_NAME: Literal["handlebars"] = "handlebars"
JINJA2_TEMPLATE_FORMAT_NAME: Literal["jinja2"] = "jinja2"

TEMPLATE_FORMAT_TYPES = Literal["semantic-kernel", "handlebars", "jinja2"]

File: nodejs/semantic-kernel/prompt-template/const.ts
/**
 * Template format name for Semantic Kernel template format.
 */
export const KERNEL_TEMPLATE_FORMAT_NAME = 'semantic-kernel' as const

/**
 * Template format name for Handlebars template format.
 */
export const HANDLEBARS_TEMPLATE_FORMAT_NAME = 'handlebars' as const

/**
 * Template format name for Jinja2 template format.
 */
export const JINJA2_TEMPLATE_FORMAT_NAME = 'jinja2' as const

/**
 * Union type for all supported template format types.
 */
export type TemplateFormatTypes = 'semantic-kernel' | 'handlebars' | 'jinja2'
